client/blocks/term-form-dialog/style.scss

```diff
@@ -20,3 +20,7 @@
 		min-width: none;
 	}
 }
+
+.term-form-dialog textarea {
+	resize: 'vertical';
```

Comment:
As-is this is not working for me in testing, I can still horizontally resize the textarea - I think you need to remove the `'` around the value so:  `resize: vertical` and it then works.